### PR TITLE
Implement basic Firehose -> ES destination

### PR DESCRIPTION
**Pull request by yknx4**
Currently Firehose doesn't have any testing so I'm not sure on how to test this new feature. But any feedback is appreciated.

We have a system where we need to stream all of our dynamodb records straight into elastic search via FIrehose.

### CODE REVIEW

File: localstack/services/firehose/firehose_api.py [hunk 1]
```diff
@@ -39,6 +43,33 @@ def put_record(stream_name, record):
...
     for dest in stream['Destinations']:
+        if 'ESDestinationDescription' in dest:
+            es_dest = dest['ESDestinationDescription']
+            es_index = es_dest['IndexName']
+            es_type = es_dest['TypeName']
+            es = connect_elasticsearch()
+            for record in records:
+                data = base64.b64decode(record['Data'])
+                unparsed_body = json.loads(data)
+                keys = unparsed_body.get('Keys')
```
Comment (whummer) [inline localstack/services/firehose/firehose_api.py]: It seems that this code is specifically targeting records coming from a DynamoDB stream, is that right? I guess we should try and keep the code in here as generic as possible, to be able to handle arbitrary records being written to the Firehose stream.

**Comment by whummer** [comment]
Thanks for this draft PR @yknx4 ! Great to see that we're working towards supporting the Firehose->Es integration. 

I added an inline comment about possibly making the logic more generic, to handle arbitrary types of records. Can you provide a few more details about your use case, and how you tested this new functionality in your environment? In any case, we should try and replicate the AWS behavior as closely as possible.

Ideally, if we could create a simple integration test to cover the new functionality. I think as a starting point we could create a fairly straight-forward test with the following steps:
1) create an Elasticsearch (ES) domain and index
2) create a firehose stream configured with the ES domain/index as destination
3) manually put some records onto the firehose delivery stream (using boto3's firehose `put_record` method)
4) verify that the entries eventually end up being written to Elasticsearch
5) optional: connect the Firehose stream to a data source like an S3 bucket, and test the end-to-end integration (putting an object to S3, verify that it lands in Elasticsearch via the delivery stream).

Please let me know what you think, any feedback welcome. I'd also be happy to help out with the implementation if necessary.

**Comment by sumitsarkar** [comment]
Not sure if I am doing it correctly, but an ElasticSearch destination never gets added to a Firehose stream. I ran commands using the CLI with localstack running on docker. I could definitely set up the S3 destination, but ElasticSearch destination gets completely ignored. No errors or anything.

**Comment by yknx4** [comment]
@sumitsarkar Indeed it is not implemented yet. This PR is the initial work towards that


@whummer thanks for your feedback. I'll update the PR with the use case and more information. And will be working on the comments soon.

**Comment by sumitsarkar** [comment]
@yknx4 Thanks for the update :) 👍

**Comment by whummer** [comment]
Thanks for updating the PR @yknx4 ! Are you still working on the tests? Otherwise, we can also get this merged as-is, and add tests later.

**Pull request merged**